app/src/main/jni/opengl/gpu_camera_adapter.cpp

```diff
@@ -211,6 +223,36 @@ struct GpuCameraAdapterImplementation : GpuCameraAdapter {
                 spec.internalFormat = GL_R8;
                 spec.format = GL_RED;
```

Comment:
Since there is no specific shader for GRAY, it will take the default and just write the rgba to single channel texture, thus taking just red channel as the gray value.

I think replicating OpenCVs conversion would be just following (https://docs.opencv.org/3.4/de/d25/imgproc_color_conversions.html):

```
float rgbaToGray(vec4 rgba) {
	return dot(rgba.rgb, vec3(0.299, 0.587, 0.114));
}

```

Reply:
Good point, this would be a nice improvement. We should also do the same for the GRAY_COMPRESSED

Reply:
Will deal with this later